R/convert_to_numeric.R

```diff
@@ -63,25 +63,19 @@ detect_to_numeric_columns <- function(scan_res) {
   checkmate::assert_data_frame(scan_res, min.rows = 1L, min.cols = 1L,
                                null.ok = FALSE)
   to_numeric <- vector(mode = "character", length = 0L)
-  for (col in scan_res[["Field_names"]]) {
-    idx <- match(col, scan_res[["Field_names"]])
-    values        <- scan_res[idx, 2L:ncol(scan_res)]
-    names(values) <- colnames(scan_res)[2L:ncol(scan_res)]
-    values        <- values[which(values > 0L)]
-    if ("missing" %in% names(values)) {
-      values <- values[-(which(names(values) == "missing"))]
-    }
-    if (length(values) == 2L && "numeric" %in% names(values) &&
-        "character" %in% names(values)) {
-      if (values[["numeric"]] > (2.0 * values[["character"]])) {
+  scan_res <- tibble::column_to_rownames(scan_res, "Field_names")
+  for (col in rownames(scan_res)) {
+    values <- unlist(scan_res[col, ])
+    values <- values[values > 0L]
+    values <- values[names(values) != "missing"]
+    if (setequal(names(values), c("numeric", "character"))) {
+      if (values[["numeric"]] > (2 * values[["character"]])) {
         to_numeric <- c(to_numeric, col)
-      } else if (values[["numeric"]] < (2L * values[["character"]])) {
+      } else if (values[["numeric"]] < (2 * values[["character"]])) {
           warning(sprintf("In '%s' column, the number of numeric values", col),
                           " is same as the number of character values",
                   call. = FALSE)
```

Comment:
The warning message here is inaccurate. @Karim-Mane, can you clarify what the intent was to help determine if the condition or the message should be fixed?

Reply:
For character columns with a certain percent of numeric values, this function will convert the column into numeric when requested by the user. Non-numeric values will then be replaced with NA.

However, there are cases where the proportion of numeric is almost similar to the proportion of character or less than 2 times, this warning is issued to alert the user. It is useful when the function is called together with other functions in a pipeable way as it is a signal for the user to take a closer look into that column.

Reply:
Yes but the message is inaccurate.

"Less than 2 times more" is not equivalent to "the same".

Reply:
I have updated the message in commit `3fed9d4`.